text/0000-bundle-add.md

```diff
@@ -0,0 +1,132 @@
+
+The purpose of this RFC is to get feedback on use cases, cli format/syntax, and feature options.
+
+# Motivation
+
+The main use case is to deploy scripts and/or improve user productivity by adding a gem to Gemfile and bundle install in one step.
+
+The question is how far to extend this feature, is there a need to support all gem options like version, source, and group. And whether or not it should support multiple gems with their respective options.
+
+This command overlaps in function with the current `bundle inject GEM` command.
+
+# Detailed design
+
+The usage of the feature can be grouped into 4 categories:
+
+## 1. add a single gem
+#### Proposed command
+```ruby
+bundle add "GEM"
+
+#examples
+bundle add "rails"
+```
+#### Description
+This is the base use case. The following steps will occur when the command is executed:
+1. It will load the current Gemfile and resolve the new GEM with it.
+2. If it fails to resolve correctly or is an invalid GEM, it will error out.
+3. If it resolves successfully, it will install the latest version of the GEM.
+4. It will add the GEM to the Gemfile with a comment and place a conservative version update requirement e.g.
+```ruby
+  # Gemfile
+  # Added at 2016-12-07 00:21:12 -0800 by username:
+  gem "GEM", "~> 1.0.0"
+```
+
+## 2. add a single gem with options
+#### Proposed command
+```ruby
+# First priority
+bundle add "GEM" [--version VERSION]
+
+# Possible full implementation
+bundle add "GEM" [--version VERSION] [--source SOURCE] [--group GROUP]
+
+#examples
+bundle add "rails" --version "~> 5.0.0"
+bundle add "rails" --version ">= 5.0, < 5.1"
+bundle add "rails" --version "~> 5.0.0" --group "development, test"
+bundle add "rails" --version "~> 5.0.0" --source "https://gems.example.com" --group "development"
+```
+#### Description
+This extends the base case allowing a user to specify optional version, source, and group.
+
+The following steps will occur when the command is executed:
+1. It will load the current Gemfile and resolve the new GEM with it.
+2. If it fails to resolve correctly or is an invalid GEM/options, it will error out.
+3. If it resolves successfully, it will install the specified or latest version of the GEM.
+4. It will add the GEM to the Gemfile with a comment and the specified options. If the version is ">=0", then it will substitute it with a conservative version update requirement.
+```ruby
+  # Gemfile
+  # Added at 2016-12-07 00:21:12 -0800 by username:
+  gem "GEM" --version "~> 5.0.0"
+  gem "GEM" --version ">= 5.0", "< 5.1"
+  gem "GEM" --version "~> 5.0.0", group: ["development", "test"]
+  gem "GEM" --version "~> 5.0.0", source: "https://gems.example.com", group: "development"
+```
+## 3. add multiple gems
+#### Proposed command
+```ruby
+bundle add "GEM1", "GEM2", "GEM3"
```

Comment:
adding specific versions or version constraints should be allowed here using the same syntax as `gem install`: for example, `bundle add GEM1:2.0 GEM2:~>1.2 GEM3:>=4.0`.